ledger/eval.go

```diff
@@ -923,58 +924,43 @@ func (eval *BlockEvaluator) GenerateBlock() (*ValidatedBlock, error) {
 type evalTxValidator struct {
 	txcache          VerifiedTxnCache
 	block            bookkeeping.Block
-	proto            config.ConsensusParams
 	verificationPool execpool.BacklogPool
 
 	ctx      context.Context
-	cf       context.CancelFunc
-	txgroups chan []transactions.SignedTxnWithAD
+	txgroups [][]transactions.SignedTxnWithAD
 	done     chan error
 }
 
 func (validator *evalTxValidator) run() {
-	for txgroup := range validator.txgroups {
-		select {
-		case <-validator.ctx.Done():
-			validator.done <- validator.ctx.Err()
-			validator.cf()
-			close(validator.done)
-			return
-		default:
-		}
-		groupNoAD := make([]transactions.SignedTxn, len(txgroup))
-		for i := range txgroup {
-			groupNoAD[i] = txgroup[i].SignedTxn
-		}
-		ctxs := verify.PrepareContexts(groupNoAD, validator.block.BlockHeader)
-
-		for gi, tx := range txgroup {
-			err := validateTransaction(tx.SignedTxn, validator.block, validator.proto, validator.txcache, ctxs[gi], validator.verificationPool)
+	defer close(validator.done)
+	verifyParams := verify.Params{
+		CurrSpecAddrs: transactions.SpecialAddresses{
+			FeeSink:     validator.block.BlockHeader.FeeSink,
+			RewardsPool: validator.block.BlockHeader.RewardsPool,
+		},
+		CurrProto: validator.block.BlockHeader.CurrentProtocol,
+	}
+	var unverifiedTxnGroups [][]transactions.SignedTxn
+	unverifiedTxnGroups = make([][]transactions.SignedTxn, len(validator.txgroups))
```

Comment:
this was probably supposed to be the 3 arg form make(type, len, cap) with len=0

Reply:
good catch.